### PR TITLE
feat(lep): non-disruptive volume-related settings

### PR DESCRIPTION
**Pull request by mantissahz**
Volume-related settings can be modified immediately and the setting controller will check if updated settings can take effect.

Ref: longhorn/longhorn#7173

### CODE REVIEW

**Comment by innobead** [comment]
@mantissahz because LEP will be reviewed and fine-tuned over time, also tech driven, so it is optional for doc review. 

Doc review should more focus on website and chart, user facing info instead.

**Comment by innobead** [comment]
Please update this to reflect the current implementation.

**Comment by innobead** [comment]
> This feature has been introduced to 1.6.0 already. After fixing the following items. We can merge it first.
> 
> * Please help review this LEP again to fix several typo and grammar issues.
> * Add [fix(setting): UX of updating danger zone settings longhorn-manager#2728](https://github.com/longhorn/longhorn-manager/pull/2728) to this LEP.

@mantissahz Why hasn't this been updated yet?

**Comment by mantissahz** [comment]
>@mantissahz Why hasn't this been updated yet?

Sorry for the late update because of focusing on multiple backup targets and 1.7.0 items.

>Please help review this LEP again to fix several typo and grammar issues.
Add https://github.com/longhorn/longhorn-manager/pull/2728 to this LEP.

Reviewed it again and updated it with the new field `Status.Applied`.

File: enhancements/20231226-support-non-disruptive-volume-related-setting-updates.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+```shell
+> kubectl -n longhorn-system get setting priority-class
+NAME             VALUE               APPLIED   AGE
+priority-class   longhorn-critical   true      5h30m
+```
+
+#### Check Setting Status By UI
+
+Users can check if the setting has been applied to Longhorn components on the Setting/General page. The Danger Zone section will display any unapplied Danger Zone settings.
+ ![](./image/danger_zone_unapplied_setting.png)
```
Comment (derekbit) [inline enhancements/20231226-support-non-disruptive-volume-related-setting-updates.md:52]: Need to update the change (`status.applied`) and checking methods to the official doc.

File: enhancements/20231226-support-non-disruptive-volume-related-setting-updates.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+        return true, false, false err
+    }
+    ```
+
+### Test plan
+
+- Test no volume attached
+    1. Fresh install Longhorn
+    1. Customize a Danger Zone setting such as setting the value of `priority-class` to be `system-cluster-critical`
```
Comment (derekbit) [inline enhancements/20231226-support-non-disruptive-volume-related-setting-updates.md:207]: Can you briefly introduce what settings should put in Danger Zone Settings?

**Review by derekbit — COMMENTED** [review]
In general, LGTM. 
The official doc is out of date. Need to update it.

File: enhancements/20231226-support-non-disruptive-volume-related-setting-updates.md (hunk 2 above)
Comment (mantissahz) [reply enhancements/20231226-support-non-disruptive-volume-related-setting-updates.md:222]: Added a section to briefly introduce Danger Zone settings.

File: enhancements/20231226-support-non-disruptive-volume-related-setting-updates.md (hunk 1 above)
Comment (mantissahz) [reply enhancements/20231226-support-non-disruptive-volume-related-setting-updates.md:67]: Updating documents.

**Comment by innobead** [comment]
> In general, LGTM. The official doc is out of date. Need to update it.

Have we finished the doc update?

**Comment by mantissahz** [comment]
>Have we finished the doc update?

Yes, it should be updated.

**Review by derekbit — APPROVED** [review]

**Pull request merged**